kubespawner/spawner.py

```diff
@@ -2158,3 +2310,22 @@ async def load_user_options(self):
                     )
                 )
             )
+
+    async def _ensure_namespace(self):
+        ns = make_namespace(self.namespace)
+        api = self.api
+        try:
+            api.create_namespace(ns)
```

Comment:
let's keep our API requests async:

```suggestion
            await gen.with_timeout(
                timedelta(seconds=self.k8s_api_request_timeout),
                self.asynchronize(api.create_namespace, ns),
            )
```

Reply:
Done!